ste/jobStatusManager.go

```diff
@@ -155,19 +181,23 @@ func (jm *jobMgr) handleStatusUpdateMessage() {
 		case <-jstm.listReq:
 			/* Display stats */
 			js.Timestamp = time.Now().UTC()
+			jstm.flagMutex.RLock() // Read-Lock
 			jstm.respChan <- *js
+			jstm.flagMutex.RUnlock()
```

Comment:
jstm.flagMutex.RLock() 
defer jstm.flagMutex.RUnlock() 
jstm.respChan <- *js

Using defer ensures that the lock is always released, even if the send operation panics or encounters an error.